modules/dag-l1/src/main/scala/org/tessellation/dag/l1/domain/consensus/block/Validator.scala

```diff
@@ -73,7 +74,37 @@ object Validator {
             if (!enoughTips) "Not enough tips" else "",
             if (!enoughTxs) "No transactions" else ""
           ).filter(_.nonEmpty).mkString(", ")
-          logger.debug(s"Cannot start own consensus: ${reason}")
+          logger.debug(s"Cannot start currency own consensus: $reason")
+        }
+    } yield res
+
+  def canStartOwnDataConsensus[F[_]: Async](
+    consensusStorage: ConsensusStorage[F],
+    nodeStorage: NodeStorage[F],
+    clusterStorage: ClusterStorage[F],
+    blockStorage: BlockStorage[F],
+    lastGlobalSnapshot: LastSnapshotStorage[F, GlobalIncrementalSnapshot, GlobalSnapshotInfo],
+    peersCount: PosInt,
+    tipsCount: PosInt
+  ): F[Boolean] =
+    for {
+      noOwnRoundInProgress <- consensusStorage.ownConsensus.get.map(_.isEmpty)
+      stateReadyForConsensus <- nodeStorage.getNodeState.map(isReadyForBlockConsensus)
+      enoughPeers <- enoughPeersForConsensus(clusterStorage, peersCount)
+      enoughTips <- enoughTipsForConsensus(blockStorage, tipsCount)
+      lastGlobalSnapshotPresent <- lastGlobalSnapshot.get.map(_.isDefined)
+
+      res = noOwnRoundInProgress && stateReadyForConsensus && enoughPeers && enoughTips && lastGlobalSnapshotPresent
+      _ <-
+        Applicative[F].whenA(!res) {
```

Comment:
Eventually `Applicative[F].unlessA(res)`. Whichever you prefer.

Reply:
Let's keep this to be in the same pattern than currency validation